src/analyses/goto_check.cpp

```diff
@@ -894,23 +894,22 @@ void goto_checkt::pointer_overflow_check(
   if(!enable_pointer_overflow_check)
     return;
 
-  if(expr.id()==ID_plus ||
-     expr.id()==ID_minus)
-  {
-    if(expr.operands().size()==2)
-    {
-      exprt overflow("overflow-"+expr.id_string(), bool_typet());
-      overflow.operands()=expr.operands();
+  if(expr.id() != ID_plus && expr.id() != ID_minus)
+    return;
 
-      add_guarded_claim(
-        not_exprt(overflow),
-        "pointer arithmetic overflow on "+expr.id_string(),
-        "overflow",
-        expr.find_source_location(),
-        expr,
-        guard);
-    }
-  }
+  if(expr.operands().size() != 2)
+    throw "pointer arithmetic expected to have exactly 2 operands";
+
```

Comment:
DATA_INVARIANT?

Reply:
I'll change it right away, but I note that it is out of sync with the rest of the code in this file. But it looks like step-by-step migration is the best route forward.